### ISSUE TITLE
Currency parsing

### ISSUE THREAD

**Issue by rxaviers** [globalizejs/globalize#364]
Implement currency parsing

Ref #238 

Depends on:
- [x] #353

**Comment by goatandsheep** [comment]
Until this feature is launched, please remove the "and parsing" from "Currency module provides currency formatting and parsing" on the README. It's extremely misleading.

**Comment by rxaviers** [comment]
True, it might have been a typo, would you like to submit a PR fix? Thanks for spotting it!

**Comment by goatandsheep** [comment]
I have an idea, actually. What if we:

1. Parse the currency string for numbers, commas, and periods: `valueString.replace(/[^\d\.\,]/g, '')`
2. Run `numberParser`
